zerver/views/public_export.py

```diff
@@ -0,0 +1,40 @@
+from datetime import timedelta
+
+from django.utils.timezone import now as timezone_now
+from django.utils.translation import ugettext as _
+from django.http import HttpResponse, HttpRequest
+
+from zerver.decorator import require_realm_admin
+from zerver.models import RealmAuditLog, UserProfile
+from zerver.lib.queue import queue_json_publish
+from zerver.lib.response import json_error, json_success
+
+@require_realm_admin
+def public_only_realm_export(request: HttpRequest, user: UserProfile) -> HttpResponse:
+    event_type = RealmAuditLog.REALM_EXPORTED
+    event_time = timezone_now()
+    realm = user.realm
+    time_delta_limit = 5
+    event_time_delta = event_time - timedelta(days=7)
+
+    # Filter based upon the number of events that have occurred in the delta
+    # If we are at the limit, the incoming request is rejected
+    limit_check = RealmAuditLog.objects.filter(event_type=event_type,
+                                               event_time__gte=event_time_delta)
```

Comment:
This query should filter on the realm!  Otherwise you're rate limiting export for everyone, whereas we want a per-organization limit.